java-bootcamp-2018-tdd/src/main/java/com/globant/controller/MainController.java

```diff
@@ -0,0 +1,193 @@
+package com.globant.controller;
+
+import java.util.Hashtable;
+
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestHeader;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RestController;
+
+import com.globant.business.shoppingCart.IShoppingCart;
+import com.globant.dbConnection.SQLConnection;
+import com.globant.dbConnection.SQLSingleton;
+import com.globant.models.ProductModel;
+import com.globant.models.RowCartModel;
+import com.globant.models.UserModel;
+
+/**
+ * main rest controller for web service
+ *
+ */
+@RestController
+public class MainController {
+	private final static Hashtable<String, String> sessions = new Hashtable<String, String>();
+	private static final String ALPHA_NUMERIC_STRING = "ABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789";
+	private static SQLConnection sql;
+
+	protected static void setUP() throws Exception {
+		sql = SQLSingleton.getInstance();
+	}
+
+	/**
+	 * login user and generate a session
+	 * 
+	 * @param user
+	 * @return the session id
+	 * @throws Exception
+	 */
+	@RequestMapping(value = "/login", method = RequestMethod.POST)
+	public String login(@RequestBody UserModel user) throws Exception {
+		UserModel userProfile = sql.findUser(user.getNickName());
+		if (userProfile != null) {
+			if (userProfile.getPassword().equals(user.getPassword())) {

```

Comment:
In real life you should compare and store passwords hashes

Reply:
How should I implement it? I save the hash in the DB? or i do the hash function when comparing?

Reply:
Not for this exercise, but in real life what you do is to save the password hash in the DB, then when the user tries to login you hash the entered password and compare the hashes.
That way the password isn't stored in plain text.
Now the hash algorithm used is SHA1, some time ago MD5 was used

Reply:
But in Java I can call a hashCode method of the String class that returns an integer, save that and use it to compare. or import a SHA1 library, I think it's not a big problem. Or is it better to leave it that way for the exercise?

Reply:
Its ok for this exercise there is no need, its just something nice to know.